libbeat/beat/monitoring.go

```diff
@@ -0,0 +1,56 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beat
+
+import "github.com/elastic/elastic-agent-libs/monitoring"
+
+type Monitoring struct {
+	namespaces *monitoring.Namespaces
+
+	InfoRegistry  *monitoring.Registry
+	StateRegistry *monitoring.Registry
+	// Previously monitoring.Default or monitoring.GetNamespace("stats")
+	StatsRegistry *monitoring.Registry
+
+	// Previously monitoring.GetNamespace("dataset")
+	InputsRegistry *monitoring.Registry
+}
+
+func NewGlobalMonitoring() Monitoring {
+	return Monitoring{
+		StatsRegistry: monitoring.Default,
+		StateRegistry: monitoring.GetNamespace("state").GetRegistry(),
+		InfoRegistry:  monitoring.GetNamespace("info").GetRegistry(),
+
+		InputsRegistry: monitoring.GetNamespace("dataset").GetRegistry(),
+	}
+}
+
+func NewMonitoring() Monitoring {
+	namespaces := monitoring.NewNamespaces()
+
+	return Monitoring{
+		namespaces: namespaces,
+
+		StatsRegistry: namespaces.Get("stats").GetRegistry(),
```

Comment:
How do you feel about turning "state", "stats", "info" and "dataset" into constants?

I know we shouldn't need them so much in the code base, but it would provide a nice place to comment them and should make it easier for a new person to tie together where they are created and where they made available in the api stats endpoints.